pkg/provisioning/config/yaml/parser_test.go

```diff
@@ -135,19 +139,19 @@ func TestParser_Warnings(t *testing.T) {
 	is.NoErr(err)
 
 	// check warnings
-	want := `{"level":"warn","component":"yaml.Parser","line":5,"column":5,"message":"field unknownField not found in type yaml.Pipeline"}
-{"level":"warn","component":"yaml.Parser","line":31,"column":15,"field":"dead-letter-queue","value":"my-plugin","message":"field dead-letter-queue was introduced in version 1.1, please update the pipeline config version"}
-{"level":"warn","component":"yaml.Parser","line":33,"column":14,"field":"dead-letter-queue","value":"bar","message":"field dead-letter-queue was introduced in version 1.1, please update the pipeline config version"}
-{"level":"warn","component":"yaml.Parser","line":34,"column":20,"field":"dead-letter-queue","value":"4","message":"field dead-letter-queue was introduced in version 1.1, please update the pipeline config version"}
-{"level":"warn","component":"yaml.Parser","line":35,"column":30,"field":"dead-letter-queue","value":"2","message":"field dead-letter-queue was introduced in version 1.1, please update the pipeline config version"}
+	want := `{"level":"warn","component":"yaml.Parser","line":5,"column":5,"message":"field unknownField not found in type v1.Pipeline"}
+{"level":"warn","component":"yaml.Parser","line":18,"column":11,"field":"processors","value":"","message":"the order of processors is non-deterministic in configuration files with version 1.x, please upgrade to version 2.x"}
+{"level":"warn","component":"yaml.Parser","line":24,"column":7,"field":"processors","value":"","message":"the order of processors is non-deterministic in configuration files with version 1.x, please upgrade to version 2.x"}
+{"level":"warn","component":"yaml.Parser","line":31,"column":7,"field":"dead-letter-queue","value":"","message":"field dead-letter-queue was introduced in version 1.1, please update the pipeline config version"}
+{"level":"warn","component":"yaml.Parser","line":52,"column":11,"field":"processors","value":"","message":"the order of processors is non-deterministic in configuration files with version 1.x, please upgrade to version 2.x"}
```

Comment:
Related to https://github.com/ConduitIO/conduit/pull/928#discussion_r1126720243: notice how we previously had 4 warnings related to `dead-letter-queue` and we now only have 1.